### PR TITLE
Sprint-7/YC-330

### PR DESCRIPTION
**Pull request by felicemcc**

### CODE REVIEW

File: permits/views.py [hunk 1]
```diff
@@ -393,6 +393,16 @@ def handle_amend_form_submission(self, form):
...
 
+        if form.instance.status == models.PermitRequest.STATUS_RECEIVED:
+            data = {
+                "subject": _("Votre annonce a été prise en compte et classée"),
+                "users_to_notify": [form.instance.author.user.email],
+                "template": "permit_request_received.txt",
+                "permit_request": form.instance,
+                "absolute_uri_func": self.request.build_absolute_uri,
+            }
+            services._send_email_notification(data)
```
Comment (T4m) [inline permits/views.py:404]: _send_email_notification is marked as a protected function. Please create a public service function or rename this one as public (without "_").

Comment (felicemcc) [reply permits/views.py]: Solved!

**Review by T4m — CHANGES_REQUESTED** [review]

File: permits/views.py [hunk 2]
```diff
@@ -1098,6 +1129,15 @@ def permit_request_classify(request, permit_request_id, approve):
...
         if classify_form.is_valid():
             classify_form.save()
+            data = {
+                "subject": _("Votre demande a été traitée et classée"),
+                "users_to_notify": [permit_request.author.user.email],
+                "template": "permit_request_classified.txt",
+                "permit_request": permit_request,
+                "absolute_uri_func": request.build_absolute_uri,
+            }
+            services._send_email_notification(data)
```
[the remark this answers is not in the document]
Comment (felicemcc) [reply permits/views.py]: the function has been renamed

File: permits/services.py [hunk 3]
```diff
@@ -854,65 +854,53 @@ def submit_permit_request(permit_request, absolute_uri_func):
...
+        data = {
+            "subject": _("Nouvelle demande"),
+            "users_to_notify": users_to_notify,
+            "template": "permit_request_submitted.txt",
+            "permit_request": permit_request,
+            "absolute_uri_func": absolute_uri_func,
+        }
+        _send_email_notification(data)
+
+        data2 = {
```
[the remark this answers is not in the document]
Comment (felicemcc) [reply permits/services.py]: data was reused

File: permits/views.py [hunk 4]
```diff
@@ -434,6 +444,27 @@ def handle_validation_form_submission(self, form):
...
+                        "Les services chargés de la validation d'une demande ont donné leur préavis"
+                    ),
+                    "users_to_notify": services._get_secretary_email(
+                        self.permit_request
+                    ),
+                    "template": "permit_request_validated.txt",
+                    "permit_request": self.permit_request,
+                    "absolute_uri_func": self.request.build_absolute_uri,
+                }
+                services._send_email_notification(data)
```
[the remark this answers is not in the document]
Comment (felicemcc) [reply permits/views.py]: Both issues solved!

File: permits/services.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (felicemcc) [reply permits/services.py]: now the double click does not send emails again...

**Pull request merged**